### PR TITLE
Sync HUD Assessments to CAS

### PR DESCRIPTION
**Pull request by eanders**
@awisema, this is not finished, but should be merge-able with backwards compatible default behavior.  We'll complete the contact changes on another branch.

### CODE REVIEW

File: app/models/concerns/arel_helper.rb [hunk 1]
```diff
@@ -155,9 +155,10 @@ def self.one_for_column(column, source_arel_table:, group_on:, direction: :desc,
       end
 
       direction = :desc unless direction.in?([:asc, :desc])
+      group_columns = Array.wrap(group_on).map { |c| source_arel_table[c] }
 
-      max_by_group = source.distinct_on(source_arel_table[group_on]).
-        order(source_arel_table[group_on], source_arel_table[column].send(direction))
+      max_by_group = source.distinct_on(group_columns).
```
Comment (awisema) [inline app/models/concerns/arel_helper.rb:160]: Should this be `*group_columns` like the next line?

File: app/models/grda_warehouse/tasks/push_clients_to_cas.rb [hunk 2]
```diff
@@ -115,6 +117,7 @@ def client_source
       GrdaWarehouse::Hud::Client.cas_active
     end
 
+    # TODO: update so all things coming from HMIS assessments or ETO get a wrapper method that checks assessments first and then client attribute
```
Comment (awisema) [inline app/models/grda_warehouse/tasks/push_clients_to_cas.rb]: Is this satisfied by the Calculator classes?

File: app/models/concerns/cas_client_data.rb [hunk 3]
```diff
@@ -0,0 +1,265 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module CasClientData
```
Comment (awisema) [inline app/models/concerns/cas_client_data.rb:7]: A possible future refactoring of this would be to move the site-specific stuff into drivers?

**Review by awisema — CHANGES_REQUESTED** [review]

File: app/models/concerns/arel_helper.rb (hunk 1 above)
Comment (eanders) [reply app/models/concerns/arel_helper.rb:160]: You would think, but that threw an error about arguments.  It seemed happier with an array.

File: app/models/grda_warehouse/tasks/push_clients_to_cas.rb (hunk 2 above)
Comment (eanders) [reply app/models/grda_warehouse/tasks/push_clients_to_cas.rb]: Yes, but maybe not completely, so I left it as a reminder.

File: app/models/concerns/cas_client_data.rb (hunk 3 above)
Comment (eanders) [reply app/models/concerns/cas_client_data.rb:7]: Yes, though, realistically, I doubt we'll ever enable drivers based on installation.

**Pull request merged**